### PR TITLE
Add vote model and associations.

### PR DESCRIPTION
**Pull request by tristandunn**
## What did we change?

- Added `Vote` model and some related associations.

## Why are we doing this?

We want to allow people to vote on projects for a Hackathon.

## Checklist

- [x] Automated Tests (check all that apply)
  - [x] Unit
- [ ] Migrations Reviewed (Zero Downtime https://ezcater.atlassian.net/l/c/F1u4298i)

### CODE REVIEW

File: app/models/vote.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Vote < ApplicationRecord
+  belongs_to :award, optional: false
+  belongs_to :event, optional: false
+  belongs_to :project, optional: false
+
+  validates :name, presence: true,
+                   uniqueness: { case_sensitive: false, scope: %i(event_id award_id) }
+end
```
Comment (daniel-patterson-ezcater) [inline app/models/vote.rb:10]: Nice!

File: db/migrate/20210929184506_create_votes.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class CreateVotes < ActiveRecord::Migration[6.0]
+  def change
+    create_table :votes do |t|
+      t.belongs_to :award, null: false
+      t.belongs_to :event, null: false
+      t.belongs_to :project, null: false
```
Comment (daniel-patterson-ezcater) [inline db/migrate/20210929184506_create_votes.rb:8]: 🎱 It _might_ be worth adding foreign keys here. However, given that these would be separate migrations, we could do it later.

File: spec/factories/votes.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :vote do
+    award
+    event
+    project
+
+    name { "Lil Sebastian" }
```
Comment (daniel-patterson-ezcater) [inline spec/factories/votes.rb:9]: 😄

**Review by daniel-patterson-ezcater — APPROVED** [review]

File: spec/factories/votes.rb (hunk 3 above)
Comment (JessSumner) [inline spec/factories/votes.rb:9]: Lil Sebastian should get double the voting power!

**Pull request merged**